src/analyze/analyze-chid.c

```
@@ -223,6 +247,87 @@ static int smbios_fields_acquire(char16_t *fields[static _CHID_SMBIOS_FIELDS_MAX
         return 0;
 }
 
+static int edid_parse(sd_device *drm_card, char16_t **ret_panel) {
+        assert(drm_card);
+        assert(ret_panel);
+        const char *edid_content;
+        size_t edid_size;
+        int r;
+
+        r = sd_device_get_sysattr_value_with_size(drm_card, "edid", &edid_content, &edid_size);
+        if (r < 0)
+                return r;
+        if (edid_size == 0)
+                return -ENODEV;
```

i would probably return a disinct error here. maybe ENODATA?

---

I'd say ENODEV fits well here. This branch is taken for video outputs which don't have anything connected to them, hence "no device"

---

The problem here is `sd_device_new_from_xyz()` also returns -ENODEV.
Please use something different, e.g. -ENXIO, which matches `ERRNO_IS_DEVICE_ABSENT()`.